torch/fx/experimental/symbolic_shapes.py

```diff
@@ -170,6 +170,41 @@ def is_concrete_int(a: Union[int, SymInt]):
 def tensor_has_hints(t):
     return all(has_hint(s) for s in t.size())
 
+def free_symbols(val: Union[SymInt, torch.Tensor]) -> Set[sympy.Symbol]:
+    if isinstance(val, SymInt):
+        return val.node.expr.free_symbols
+    elif isinstance(val, int):
+        return set()
+    elif isinstance(val, torch.Tensor):
+        r = set()
+        for s in val.size():
+            r |= free_symbols(s)
+        for s in val.stride():
+            r |= free_symbols(s)
+        r |= free_symbols(val.storage_offset())
+        return r
+    else:
+        raise AssertionError(f"cannot compute free_symbols of {val}")
+
+# WARNING: Don't use this on Dynamo produced graphs, they don't have meta
+# setup!
+def is_symbol_binding_fx_node(node) -> Optional[sympy.Symbol]:
```

Comment:
Yes, this is a refactor I was thinking about. One good start would be to change Dynamo to use "val" instead of "example_input". Another would be top stop deleting example_input after we're done compiling.